Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -182,7 +182,7 @@ private static void OnCarouselPropertyChanged(DependencyObject d, DependencyProp
 
             Carousel carouselControl = (Carousel)d;
 
-            if (e.Property == SelectedIndexProperty)
+            if (e.Property == SelectedIndexProperty && (int)e.NewValue < 0)
```

Comment:
My bad. I marged two versions of the same code. I will propose next commit that will fix it. Also I will  add your next suggestion to it.